### PR TITLE
Project restructure

### PR DESCRIPTION
**Pull request by ganttastic**
Restructured project to match standard across all BR open source frameworks.

### CODE REVIEW

File: .swiftlint.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Set max line length before warning (default is 120)
+line_length: 245
+
+disabled_rules:
+  - trailing_whitespace # Disables SwiftLint complaining about whitespace characters on empty lines
+  - todo # Disables auto-warning of TODO statements
+
+identifier_name:
+  excluded:
```
Comment (tylermilner) [inline .swiftlint.yml]: Are these exclusions needed for this project? If not, let's go ahead and clean them up.

File: Examples/SessionTools-iOSExample/Info.plist [hunk 2]
```diff
@@ -15,7 +15,7 @@
 	<key>CFBundlePackageType</key>
 	<string>APPL</string>
 	<key>CFBundleShortVersionString</key>
-	<string>1.0.2</string>
+	<string>1.0.3</string>
```
Comment (tylermilner) [inline Examples/SessionTools-iOSExample/Info.plist]: Let's keep the version set to `1.0.2` since we'll do a separate PR for the actual release.

File: SessionTools.podspec [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 Pod::Spec.new do |s|
     s.name             = 'SessionTools'
-    s.version          = '1.0.2'
+    s.version          = '1.0.3'
```
Comment (tylermilner) [inline SessionTools.podspec]: Similar comment about version number.

File: Tests/Mocks/MockStorageContainer.swift [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import Foundation
+import SessionTools
+
+/// An in memory container for testing.
+public class MockStorageContainer<T: Codable>: SessionContainer {
+    
+    private var localStorage: [String: Data]
+    
+    public init() {
+        localStorage = [:]
```
Comment (tylermilner) [inline Tests/Mocks/MockStorageContainer.swift]: Any reason for assigning the initial value of `localStorage` here instead of just assigning it in the property declaration above?

File: Tests/Mocks/MockStorageContainer.swift [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+//  SessionTools-iOSTests
+//
+//  Copyright © 2018 Bottle Rocket Studios. All rights reserved.
+//
+
+import Foundation
+import SessionTools
+
+/// An in memory container for testing.
+public class MockStorageContainer<T: Codable>: SessionContainer {
```
Comment (tylermilner) [inline Tests/Mocks/MockStorageContainer.swift]: Since this is in the test target, I don't think we need it to be `public` (same applies to its methods).

File: .swiftlint.yml (hunk 1 above)
Comment (ganttastic) [reply .swiftlint.yml]: I removed all except for the trailing whitespace rule since Xcode adds those by default.

File: Examples/SessionTools-iOSExample/Info.plist (hunk 2 above)
Comment (ganttastic) [reply Examples/SessionTools-iOSExample/Info.plist]: Sounds good!

File: SessionTools.podspec (hunk 3 above)
Comment (ganttastic) [reply SessionTools.podspec]: Similar comment as above. :D

File: Tests/Mocks/MockStorageContainer.swift (hunk 4 above)
Comment (ganttastic) [reply Tests/Mocks/MockStorageContainer.swift]: I was thinking in the future we might want to pass in data for testing, but nothing uses that now, so I've move it up to the property declaration.

File: Tests/Mocks/MockStorageContainer.swift (hunk 5 above)
Comment (ganttastic) [reply Tests/Mocks/MockStorageContainer.swift]: Agreed. Removed public.
